tests/test_immutable_morphology.cpp

```diff
@@ -0,0 +1,386 @@
+                               std::vector<morphio::floatType>{20.0, 30.0, 40.0, 50.0},
+                               0.01));
+    relativePathLength = child.relativePathLengths();
+
+#ifdef MORPHIO_USE_DOUBLE
+    REQUIRE(array_almost_equal(std::vector<morphio::floatType>(relativePathLength.begin(),
+                                                               relativePathLength.end()),
+                               std::vector<morphio::floatType>{0.6, 0.7, 0.8, 0.9},
+                               0.01));
+#else
+    REQUIRE(array_almost_equal(std::vector<morphio::floatType>(relativePathLength.begin(),
+                                                               relativePathLength.end()),
+                               std::vector<morphio::floatType>{0.6f, 0.7f, 0.8f, 0.9f},
+                               0.01));
+#endif
+
+    neuriteSectionIds = child.neuriteSectionIds();
+    REQUIRE(array_almost_equal(std::vector<morphio::floatType>(neuriteSectionIds.begin(),
+                                                               neuriteSectionIds.end()),
+                               std::vector<morphio::floatType>{3.0, 4.0, 4.0, 5.0},
+                               0.01));
+    rootSection = mito.rootSections().at(1);
+    diameters = rootSection.diameters();
+    REQUIRE(array_almost_equal(std::vector<morphio::floatType>(diameters.begin(), diameters.end()),
+                               std::vector<morphio::floatType>{5.0, 6.0, 7.0, 8.0},
+                               0.01));
+    relativePathLength = rootSection.relativePathLengths();
+
+#ifdef MORPHIO_USE_DOUBLE
+    REQUIRE(array_almost_equal(std::vector<morphio::floatType>(relativePathLength.begin(),
+                                                               relativePathLength.end()),
+                               std::vector<morphio::floatType>{0.6, 0.7, 0.8, 0.9},
+                               0.01));
+#else
+    REQUIRE(array_almost_equal(std::vector<morphio::floatType>(relativePathLength.begin(),
+                                                               relativePathLength.end()),
+                               std::vector<morphio::floatType>{0.6f, 0.7f, 0.8f, 0.9f},
+                               0.01));
+#endif
+
+    neuriteSectionIds = rootSection.neuriteSectionIds();
+    REQUIRE(array_almost_equal(std::vector<morphio::floatType>(neuriteSectionIds.begin(),
+                                                               neuriteSectionIds.end()),
+                               std::vector<morphio::floatType>{0.0, 1.0, 1.0, 2.0},
+                               0.01));
+    REQUIRE(rootSection.children().empty());
+}
+
+
+TEST_CASE("endoplasmic_reticulum", "[immutableMorphology]") {
+    morphio::Morphology morph = morphio::Morphology("data/h5/v1/endoplasmic-reticulum.h5");
+    morphio::EndoplasmicReticulum er = morph.endoplasmicReticulum();
+    REQUIRE(er.sectionIndices() == std::vector<uint32_t>{1, 4, 5});
+#ifdef MORPHIO_USE_DOUBLE
```

Comment:
can you avoid the `#ifdef` by constructing the expected values with `morphio::floatType`?

Reply:
No due to the Werror + Wdouble-promotion + Wuseless-cast altogether I am quite force to do that. 

With gcc, if I am using  MORPHIO_USE_DOUBLE then morphio::floatType(0.42f)  works but morphio::floatType(0.42) throws a Wuseless-cast.

If I am not using  MORPHIO_USE_DOUBLE then morphio::floatType(0.42f) throws a Wuseless-cast  the but works with morphio::floatType(0.42). 

And I did not want to remove the warnings.

Reply:
ok, I guess we can templetize it then, sigh

Reply:
I am not sure what you want to templetize here